test/ts/VNode-test.tsx

```diff
@@ -116,6 +117,19 @@ describe('VNode TS types', () => {
 		const types = toChildArray(comp.props.children).map(getDisplayType);
 		expect(types).to.deep.equal(['a', '2', 'b', 'c', 'TestComp']);
 	});
+
+	it('component should work with cloneElement', () => {
+		const comp: VNode = (
+			<SimpleComponent>
+				<div>child 1</div>
+			</SimpleComponent>
+		);
+		const clone = cloneElement(comp);
```

Comment:
```suggestion
		const clone: VNode = cloneElement(comp);
```

I think this is good to type-check since this would throw if cloneElement would return an `Element`